test/interp/return-call-indirect.txt

```diff
@@ -0,0 +1,58 @@
+;;; TOOL: run-interp
+(module
+  (type $iii_i (func (param i32 i32 i32)(result i32)))
+  (table anyfunc (elem $facInd))
+
+  (func (export "call") (result i32)
+    i32.const 1
+    i64.const 2
+    f32.const 3
+    f64.const 4
+    call $helper)
+
+  (func $helper (param i32 i64 f32 f64) (result i32)
+    get_local 1
+    i32.wrap/i64
+    get_local 0
+    i32.add
+    get_local 2
+    i32.trunc_s/f32
+    i32.add
+    get_local 3
+    i32.trunc_s/f64
+    i32.add
+    return)
+
+  (func (export "facInd10") (result i32)
+    i32.const 10
+    i32.const 1
+    i32.const 0
+    i32.const 0
+    call_indirect (type $iii_i)
+    return)
+
+(;; Tail call version of factorial, using indirect call ;;)
+(;; fac(Ix,So) => Ix==0?So:fac(Ix-1,So*Ix) ;;)
+  (func $facInd (type $iii_i)
+    get_local 0
+    i32.const 0
+    i32.gt_s
+    if (result i32)
+      get_local 0
+      i32.const 1
+      i32.sub
+      get_local 1
+      get_local 0
+      i32.mul
+      get_local 2
```

Comment:
local 2 seems to just be forwarding through 0

Reply:
I don't understand this.
The third argument is the function index, not the same as either the first or third argument.
Maybe I am missing something here?

Reply:
Ah. You could. But the way I did it was kind of deliberate. I was imagining that I was implementing a functional programming language :)